mediacapture-streams/MediaDevices-IDL-enumerateDevices.html

```diff
@@ -0,0 +1,68 @@
+<!doctype html>
+<html>
+<head>
+<title>enumerateDevices: test that enumerateDevices is present</title>
+<link rel="author" title="Dr Alex Gouaillard" href="mailto:agouaillard@gmail.com"/>
+<link rel="help" href="http://w3c.github.io/mediacapture-main/getusermedia.html#methods-2">
+<link rel="help" href="http://w3c.github.io/mediacapture-main/getusermedia.html#device-info">
+<link rel="help" href="http://w3c.github.io/mediacapture-main/getusermedia.html#idl-def-MediaDeviceKind">
+<meta name='assert' content='Check that the enumerateDevices() method is present.'/>
+</head>
+<body>
+<h1 class="instructions">Description</h1>
+<p class="instructions">This test checks for the presence of the
+<code>navigator.mediaDevices.enumerateDevices()</code> method.</p>
+<div id='log'></div>
+<script src=/resources/testharness.js></script>
+<script src=/resources/testharnessreport.js></script>
+<script src=/resources/WebIDLParser.js></script>
+<script src=/resources/idlharness.js></script>
+<script>
+"use strict";
+
+var _mediaInfo = null;;
+
+var MDI_idl = new IdlArray();
+
+MDI_idl.add_idls("[Exposed=Window]\
+  interface MediaDeviceInfo {\
+    readonly        attribute DOMString       deviceId;\
+    readonly        attribute MediaDeviceKind kind;\
+    readonly        attribute DOMString       label;\
+    readonly        attribute DOMString       groupId;\
+    serializer = {attribute};\
+};");
+
+MDI_idl.add_idls('\
+  enum MediaDeviceKind {\
+    "audioinput",\
+    "audiooutput",\
+    "videoinput"\
+};');
+
+test(function(){
+  assert_true(undefined !== navigator.mediaDevices.enumerateDevices, "navigator.mediaDevices.enumerateDevices exists");
+  promise_test(function() {
+    return navigator.mediaDevices.enumerateDevices()
+      .then(function(list) {
+        if( list.length > 0 ) {
+          _mediaInfo = list[0];
+          MDI_idl.add_objects({MediaDeviceInfo: ["_mediaInfo"]});
+        }
```

Comment:
maybe add `else { assert_unreached();}` if list is empty (otherwise, the test would pass in a browser that doesn't implement the interface but has no connected media device)